### ISSUE TITLE
Failing to change types of target schema (TypeScript)

### ISSUE THREAD

**Issue by moltar** [nobrainr/morphism#155]
**Describe the bug**

It seems to be impossible to change types on the target schema. The value assumes the same type as the source.

**To Reproduce**

```ts
import { StrictSchema } from 'morphism'

interface SchemaSource {
  foo: string
}

interface SchemaTarget {
  foo: number
}

const schema: StrictSchema<SchemaTarget, SchemaSource> = {
  foo: {
    path: 'foo',
    fn: (val) => {
      return Number(val)
    }
  }
}
```

**Expected behavior**

Allow changing type on the target schema.

**Screenshots**

<img width="374" alt="image" src="https://user-images.githubusercontent.com/491247/70802837-d7cad580-1db2-11ea-9e05-8ffab1851021.png">

```
(property) foo: string | string[] | ((iteratee: SchemaSource, source: SchemaSource[], target: number) => number) | "foo"[] | ActionSelector<SchemaSource, SchemaTarget> | StrictSchema<...>
Type '{ path: string; fn: (val: any) => number; }' is not assignable to type 'string | string[] | ((iteratee: SchemaSource, source: SchemaSource[], target: number) => number) | "foo"[] | ActionSelector<SchemaSource, SchemaTarget> | StrictSchema<...>'.
  Type '{ path: string; fn: (val: any) => number; }' is not assignable to type 'string'.ts(2322)
Untitled-1(8, 3): The expected type comes from property 'foo' which is declared here on type 'StrictSchema<SchemaTarget, SchemaSource>'
```

**Additional context**

❯ tsc -v
Version 3.6.4

```
    "morphism": "1.12.2",
```

**Comment by moltar** [comment]
Seems to work with ActionFunction pattern:

```
const schema: StrictSchema<SchemaTarget, SchemaSource> = {
  foo: (iteratee, source, destination) => {
    return Number(iteratee.foo)
  }
}
```

No TypeScript errors.

**Comment by emyann** [comment]
@moltar Thank you for reporting this issue! This should have been fixed in [`Morphism v1.12.3`](https://github.com/nobrainr/morphism/releases/tag/v1.12.3)!

Please find a live example here: https://repl.it/@yrnd1/BurlywoodTomatoWorker

**Comment by moltar** [comment]
Thanks, Yann!
